packages/admin/src/implementation.ts

```diff
@@ -0,0 +1,233 @@
+
+  return new Promise<string>((resolve) =>
+    airnodeRrp.once(filter, (airnodeId) => {
+      resolve(airnodeId);
+    })
+  );
+}
+
+export async function deriveEndpointId(oisTitle: string, endpointName: string) {
+  return ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['string'], [`${oisTitle}_${endpointName}`]));
+}
+
+export async function clientAddressToNoRequests(airnodeRrp: AirnodeRrp, clientAddress: string) {
+  return (await airnodeRrp.clientAddressToNoRequests(clientAddress)).toString();
+}
+
+export async function getAirnodeParameters(airnodeRrp: AirnodeRrp, airnodeId: string) {
+  const ethersParams = await airnodeRrp.getAirnodeParametersAndBlockNumber(airnodeId);
+
+  // remove array parameters from ethers response
+  const { admin, authorizers, xpub, blockNumber } = ethersParams;
+  const params = { admin, authorizers, xpub, blockNumber };
+  assertAllParamsAreReturned(params, ethersParams);
+
+  // cast ethers BigNumber for portability
+  return { ...params, blockNumber: blockNumber.toString() };
+}
+
+export async function getTemplate(airnodeRrp: AirnodeRrp, templateId: string) {
+  const ethersTemplate = await airnodeRrp.getTemplate(templateId);
+
+  // remove array parameters from ethers response
+  const { airnodeId, endpointId, parameters } = ethersTemplate;
+  const template = { airnodeId, endpointId, parameters };
+  assertAllParamsAreReturned(template, ethersTemplate);
+
+  return template;
+}
+
+export async function getTemplates(airnodeRrp: AirnodeRrp, templateIds: string[]) {
+  const ethersTemplate = await airnodeRrp.getTemplates(templateIds);
+
+  // remove array parameters from ethers response
+  const { airnodeIds, endpointIds, parameters } = ethersTemplate;
+  const templates = { airnodeIds, endpointIds, parameters };
+  assertAllParamsAreReturned(templates, ethersTemplate);
+
+  const formattedTemplates = airnodeIds.map((_, i) => ({
+    airnodeId: airnodeIds[i],
+    endpointId: endpointIds[i],
+    parameters: parameters[i],
+  }));
+  return formattedTemplates;
+}
+
+export function requesterIndexToAdmin(airnodeRrp: AirnodeRrp, requesterIndex: BigNumberish) {
+  return airnodeRrp.requesterIndexToAdmin(requesterIndex);
+}
+
+export function requesterIndexToClientAddressToEndorsementStatus(
+  airnodeRrp: AirnodeRrp,
+  requesterIndex: BigNumberish,
+  clientAddress: string
+) {
+  return airnodeRrp.requesterIndexToClientAddressToEndorsementStatus(requesterIndex, clientAddress);
+}
+
+export async function requesterIndexToNextWithdrawalRequestIndex(airnodeRrp: AirnodeRrp, requesterIndex: BigNumberish) {
+  const requestsCount = await airnodeRrp.requesterIndexToNextWithdrawalRequestIndex(requesterIndex);
+  return requestsCount.toString();
+}
+
+export interface FulfillWithdrawalReturnValue {
+  airnodeId: string;
+  requesterIndex: string;
+  designatedWallet: string;
+  destination: string;
+  amount: string;
+  withdrawalRequestId: string;
+}
+
+export async function fulfilWithdrawal(
```

Comment:
Typo

Reply:
Fulfill is such a stupid word. So many times I write it as `fullfil` because of `full`.
And actually both `fulfiill` and `fulfil` are gramatically correct.